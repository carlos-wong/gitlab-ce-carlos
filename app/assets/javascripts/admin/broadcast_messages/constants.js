import { s__ } from '~/locale';

export const TYPE_BANNER = 'banner';
export const TYPE_NOTIFICATION = 'notification';

export const TYPES = [
  { value: TYPE_BANNER, text: s__('BroadcastMessages|Banner') },
  { value: TYPE_NOTIFICATION, text: s__('BroadcastMessages|Notification') },
];

export const THEMES = [
  { value: 'indigo', text: s__('BroadcastMessages|Indigo') },
  { value: 'light-indigo', text: s__('BroadcastMessages|Light Indigo') },
  { value: 'blue', text: s__('BroadcastMessages|Blue') },
  { value: 'light-blue', text: s__('BroadcastMessages|Light Blue') },
  { value: 'green', text: s__('BroadcastMessages|Green') },
  { value: 'light-green', text: s__('BroadcastMessages|Light Green') },
  { value: 'red', text: s__('BroadcastMessages|Red') },
  { value: 'light-red', text: s__('BroadcastMessages|Light Red') },
  { value: 'dark', text: s__('BroadcastMessages|Dark') },
  { value: 'light', text: s__('BroadcastMessages|Light') },
];

export const NEW_BROADCAST_MESSAGE = {
  message: '',
  broadcastType: TYPES[0].value,
  theme: THEMES[0].value,
  dismissable: false,
  targetPath: '',
  targetAccessLevels: [],
  startsAt: new Date(),
  endsAt: new Date(),
};
