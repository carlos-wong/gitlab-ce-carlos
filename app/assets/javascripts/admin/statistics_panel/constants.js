import { s__ } from '~/locale';

const statisticsLabels = {
  forks: s__('AdminStatistics|Forks'),
  issues: s__('AdminStatistics|Issues'),
  mergeRequests: s__('AdminStatistics|Merge Requests'),
  notes: s__('AdminStatistics|Notes'),
  snippets: s__('AdminStatistics|Snippets'),
  sshKeys: s__('AdminStatistics|SSH Keys'),
  milestones: s__('AdminStatistics|Milestones'),
  activeUsers: s__('AdminStatistics|Active Users'),
};

export default statisticsLabels;
