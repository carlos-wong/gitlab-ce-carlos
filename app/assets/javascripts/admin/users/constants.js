import { s__, __ } from '~/locale';

export const USER_AVATAR_SIZE = 32;

export const LENGTH_OF_USER_NOTE_TOOLTIP = 100;

export const I18N_USER_ACTIONS = {
  edit: __('Edit'),
  userAdministration: s__('AdminUsers|User administration'),
  unlock: __('Unlock'),
  block: s__('AdminUsers|Block'),
  unblock: s__('AdminUsers|Unblock'),
  approve: s__('AdminUsers|Approve'),
  reject: s__('AdminUsers|Reject'),
  deactivate: s__('AdminUsers|Deactivate'),
  activate: s__('AdminUsers|Activate'),
  ldapBlocked: s__('AdminUsers|Cannot unblock LDAP blocked users'),
  delete: s__('AdminUsers|Delete user'),
  deleteWithContributions: s__('AdminUsers|Delete user and contributions'),
  ban: s__('AdminUsers|Ban user'),
  unban: s__('AdminUsers|Unban user'),
};
