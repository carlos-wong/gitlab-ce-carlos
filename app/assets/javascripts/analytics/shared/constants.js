import { masks } from 'dateformat';
import { s__ } from '~/locale';

export const DATE_RANGE_LIMIT = 180;
export const OFFSET_DATE_BY_ONE = 1;
export const PROJECTS_PER_PAGE = 50;

const { isoDate, mediumDate } = masks;
export const dateFormats = {
  isoDate,
  defaultDate: mediumDate,
  defaultDateTime: 'mmm d, yyyy h:MMtt',
  month: 'mmmm',
};

// Some content is duplicated due to backward compatibility.
// It will be removed with https://gitlab.com/gitlab-org/gitlab/-/issues/350614 in 14.9
export const METRICS_POPOVER_CONTENT = {
  lead_time: {
    description: s__('ValueStreamAnalytics|Median time from issue created to issue closed.'),
  },
  cycle_time: {
    description: s__(
      "ValueStreamAnalytics|Median time from the earliest commit of a linked issue's merge request to when that issue is closed.",
    ),
  },
  lead_time_for_changes: {
    description: s__(
      'ValueStreamAnalytics|Median time between merge request merge and deployment to a production environment for all MRs deployed in the given time period.',
    ),
  },
  issues: { description: s__('ValueStreamAnalytics|Number of new issues created.') },
  deploys: { description: s__('ValueStreamAnalytics|Total number of deploys to production.') },
  deployment_frequency: {
    description: s__('ValueStreamAnalytics|Average number of deployments to production per day.'),
  },
  commits: {
    description: s__('ValueStreamAnalytics|Number of commits pushed to the default branch'),
  },
  time_to_restore_service: {
    description: s__(
      'ValueStreamAnalytics|Median time an incident was open on a production environment in the given time period.',
    ),
  },
  change_failure_rate: {
    description: s__(
      'ValueStreamAnalytics|Percentage of deployments that cause an incident in production.',
    ),
  },
};

const KEY_METRICS_TITLE = s__('ValueStreamAnalytics|Key metrics');
const KEY_METRICS_KEYS = ['lead_time', 'cycle_time', 'issues', 'commits', 'deploys'];

const DORA_METRICS_TITLE = s__('ValueStreamAnalytics|DORA metrics');
const DORA_METRICS_KEYS = [
  'deployment_frequency',
  'lead_time_for_changes',
  'time_to_restore_service',
  'change_failure_rate',
];

export const VSA_METRICS_GROUPS = [
  { key: 'key_metrics', title: KEY_METRICS_TITLE, keys: KEY_METRICS_KEYS },
  { key: 'dora_metrics', title: DORA_METRICS_TITLE, keys: DORA_METRICS_KEYS },
];
