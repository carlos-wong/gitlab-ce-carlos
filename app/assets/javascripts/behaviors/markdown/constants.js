// https://prosemirror.net/docs/ref/#model.ParseRule.priority
export const DEFAULT_PARSE_RULE_PRIORITY = 50;
export const HIGHER_PARSE_RULE_PRIORITY = 1 + DEFAULT_PARSE_RULE_PRIORITY;
