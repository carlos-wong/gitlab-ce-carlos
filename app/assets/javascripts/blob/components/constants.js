import { __ } from '~/locale';

export const BTN_COPY_CONTENTS_TITLE = __('Copy file contents');
export const BTN_RAW_TITLE = __('Open raw');
export const BTN_DOWNLOAD_TITLE = __('Download');

export const SIMPLE_BLOB_VIEWER = 'simple';
export const SIMPLE_BLOB_VIEWER_TITLE = __('Display source');

export const RICH_BLOB_VIEWER = 'rich';
export const RICH_BLOB_VIEWER_TITLE = __('Display rendered file');
