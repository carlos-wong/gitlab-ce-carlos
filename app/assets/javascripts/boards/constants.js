export const ListType = {
  assignee: 'assignee',
  milestone: 'milestone',
  backlog: 'backlog',
  closed: 'closed',
  label: 'label',
};

export default {
  ListType,
};
