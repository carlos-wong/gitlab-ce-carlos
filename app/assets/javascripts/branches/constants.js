export const SIDES = {
  full: 'full',
  left: 'left',
  right: 'right',
};
export const MAX_COMMIT_COUNT = 1000;
