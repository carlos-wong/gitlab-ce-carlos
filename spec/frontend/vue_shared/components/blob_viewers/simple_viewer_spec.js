import { shallowMount } from '@vue/test-utils';
import SimpleViewer from '~/vue_shared/components/blob_viewers/simple_viewer.vue';
import { HIGHLIGHT_CLASS_NAME } from '~/vue_shared/components/blob_viewers/constants';

describe('Blob Simple Viewer component', () => {
  let wrapper;
  const contentMock = `<span id="LC1">First</span>\n<span id="LC2">Second</span>\n<span id="LC3">Third</span>`;

  function createComponent(content = contentMock) {
    wrapper = shallowMount(SimpleViewer, {
      propsData: {
        content,
      },
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  it('does not fail if content is empty', () => {
    const spy = jest.spyOn(window.console, 'error');
    createComponent('');
    expect(spy).not.toHaveBeenCalled();
  });

  describe('rendering', () => {
    beforeEach(() => {
      createComponent();
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders exactly three lines', () => {
      expect(wrapper.findAll('.js-line-number')).toHaveLength(3);
    });

    it('renders the content without transformations', () => {
      expect(wrapper.html()).toContain(contentMock);
    });
  });

  describe('functionality', () => {
    const scrollIntoViewMock = jest.fn();
    HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;

    beforeEach(() => {
      window.location.hash = '#LC2';
      createComponent();
    });

    afterEach(() => {
      window.location.hash = '';
    });

    it('scrolls to requested line when rendered', () => {
      const linetoBeHighlighted = wrapper.find('#LC2');
      expect(scrollIntoViewMock).toHaveBeenCalled();
      expect(wrapper.vm.highlightedLine).toBe(linetoBeHighlighted.element);
      expect(linetoBeHighlighted.classes()).toContain(HIGHLIGHT_CLASS_NAME);
    });

    it('switches highlighting when another line is selected', () => {
      const currentlyHighlighted = wrapper.find('#LC2');
      const hash = '#LC3';
      const linetoBeHighlighted = wrapper.find(hash);

      expect(wrapper.vm.highlightedLine).toBe(currentlyHighlighted.element);

      wrapper.vm.scrollToLine(hash);

      return wrapper.vm.$nextTick(() => {
        expect(wrapper.vm.highlightedLine).toBe(linetoBeHighlighted.element);
        expect(currentlyHighlighted.classes()).not.toContain(HIGHLIGHT_CLASS_NAME);
        expect(linetoBeHighlighted.classes()).toContain(HIGHLIGHT_CLASS_NAME);
      });
    });
  });
});
