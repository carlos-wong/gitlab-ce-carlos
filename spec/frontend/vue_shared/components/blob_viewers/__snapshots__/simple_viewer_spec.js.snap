// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blob Simple Viewer component rendering matches the snapshot 1`] = `
<div
  class="file-content code js-syntax-highlight qa-file-content"
>
  <div
    class="line-numbers"
  >
    <a
      class="diff-line-num js-line-number"
      data-line-number="1"
      href="#LC1"
      id="L1"
    >
      <gl-icon-stub
        name="link"
        size="12"
      />
      
      1
    
    </a>
    <a
      class="diff-line-num js-line-number"
      data-line-number="2"
      href="#LC2"
      id="L2"
    >
      <gl-icon-stub
        name="link"
        size="12"
      />
      
      2
    
    </a>
    <a
      class="diff-line-num js-line-number"
      data-line-number="3"
      href="#LC3"
      id="L3"
    >
      <gl-icon-stub
        name="link"
        size="12"
      />
      
      3
    
    </a>
  </div>
   
  <div
    class="blob-content"
  >
    <pre
      class="code highlight"
    >
      <code
        id="blob-code-content"
      >
        <span
          id="LC1"
        >
          First
        </span>
        

        <span
          id="LC2"
        >
          Second
        </span>
        

        <span
          id="LC3"
        >
          Third
        </span>
      </code>
    </pre>
  </div>
</div>
`;
